src/unstructured_client/_hooks/custom/split_pdf_hook.py

```diff
@@ -154,27 +158,39 @@ def before_request(
         limiter = asyncio.Semaphore(concurrency_level)
 
         pdf = PdfReader(io.BytesIO(file.content))
+
+        page_range_start, page_range_end = form_utils.get_page_range(
+            form_data,
+            key=PARTITION_FORM_PAGE_RANGE_KEY,
+            max_pages=len(pdf.pages),
+        )
+
+        page_count = min(len(pdf.pages), page_range_end - page_range_start + 1)
+        logger.info(
+            "Splitting pages %d to %d (%d total)",
+            page_range_start,
+            page_range_end,
+            page_count,
+        )
+
         split_size = get_optimal_split_size(
-            num_pages=len(pdf.pages), concurrency_level=concurrency_level
+            num_pages=page_count, concurrency_level=concurrency_level
         )
         logger.info("Determined optimal split size of %d pages.", split_size)
 
-        if split_size >= len(pdf.pages):
+        # If the doc is small enough, and we aren't slicing it with a page range:
+        # do not split, just continue with the original request
+        if split_size >= page_count and page_count == len(pdf.pages):
             logger.info(
                 "Document has too few pages (%d) to be split efficiently. Partitioning without split.",
-                len(pdf.pages),
+                page_count,
             )
             return request
 
-        pages = pdf_utils.get_pdf_pages(pdf, split_size)
-        logger.info(
-            "Document split into %d, %d-paged sets.",
-            math.ceil(len(pdf.pages) / split_size),
-            split_size,
-        )
+        pages = pdf_utils.get_pdf_pages(pdf, split_size=split_size, page_start=page_range_start, page_end=page_range_end)
         logger.info(
             "Partitioning %d, %d-paged sets.",
-            math.ceil(len(pdf.pages) / split_size),
+            math.ceil(page_count / split_size),
```

Comment:
I'd introduce a variable like `chunk_no` for this or change the log message to e.g. `Paritioning %d pages chunks, %d-paged sets"